server/src/main/java/org/elasticsearch/index/mapper/ParseContext.java

```diff
@@ -131,8 +131,12 @@ public IndexableField getByKey(Object key) {
         public final String[] getValues(String name) {
             List<String> result = new ArrayList<>();
             for (IndexableField field : fields) {
-                if (field.name().equals(name) && field.stringValue() != null) {
-                    result.add(field.stringValue());
+                if (field.name().equals(name)) {
+                    if (field.stringValue() != null) {
+                        result.add(field.stringValue());
+                    } else if (field.binaryValue() != null) { // KeywordFieldType
```

Comment:
We cannot use the binary value without checking the type of the field. It could be a real `binary` field that doesn't store utf8 bytes. Since this function is only used in the terms vector service, can you move it there and adds a check to extract binary values only if the field is of type `keyword` ?

Reply:
The method is called as you said only by TermVectorService and it's guarded by a isValidField:
https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/index/termvectors/TermVectorsService.java#L304 but I agree, it makes sense to move the method close to this so it's visible.